op-node/node/node.go

```diff
@@ -434,11 +437,36 @@ func (n *OpNode) initL2(ctx context.Context, cfg *Config) error {
 	} else {
 		n.safeDB = safedb.Disabled
 	}
+
+	if cfg.Rollup.ChainOpConfig == nil {
+		chainCfg, err := loadOrFetchChainConfig(ctx, cfg.Rollup.L2ChainID, rpcClient)
+		if err != nil {
+			return fmt.Errorf("failed to load or fetch chain config for id %v: %w", cfg.Rollup.L2ChainID, err)
+		}
+		cfg.Rollup.ChainOpConfig = chainCfg.Optimism
+	}
```

Comment:
We could also use the default config, which is almost always used, and print a warning that the default is used. If the default is wrong, it's only consolidation that will fail and so node operators would notice and could fix it by adding the parameters to the `rollup.Config`.

Reply:
I think it's better to not assume configuration. Node operators can update their `rollup.json` pretty trivially if they see a clear error on startup.